test/test_stateless.py

```diff
@@ -146,11 +149,34 @@ def test_reparametrized_module_change_parametrization_original(self):
         parameters = {'l1.parametrizations.weight.original': torch.nn.Parameter(torch.tensor([[1.0]])),
                       'l1.bias': torch.tensor([0.0]),
                       'buffer': torch.tensor([0.0])}
-        res = torch.nn.utils._stateless.functional_call(module, parameters, x)
+        res = stateless.functional_call(module, parameters, x)
         self.assertEqual(x, res)
         # verify that the spectral normalization is still applied
         self.assertTrue('l1.parametrizations.weight.original' in dict(module.named_parameters()))
         self.assertEqual(orig_sn_weight, module.l1.weight)
 
+class TestStatelessDeprecation(TestCase):
+    def test_private_stateless_warns(self):
+        script = """
+import torch
+import warnings
+
+with warnings.catch_warnings(record=True) as w:
+    from torch.nn.utils import _stateless
+
+exit(len(w))
+"""
+        try:
+            subprocess.check_output(
+                [sys.executable, '-W', 'all', '-c', script],
+                stderr=subprocess.STDOUT,
+                # On Windows, opening the subprocess with the default CWD makes `import torch`
+                # fail, so just set CWD to this script's directory
+                cwd=os.path.dirname(os.path.realpath(__file__)),)
+        except subprocess.CalledProcessError as e:
+            self.assertEqual(e.returncode, 1)
+        else:
+            self.assertTrue(False, "No warning was raised.")
```

Comment:
this is to avoid how we can't import _stateless multiple times?

Reply:
This is because the exit code is the number of warning raised. But if 0 were raised, then the process exited "correctly" and no error was thrown above.
This is just to make sure that this case also raises an error.